PittAPI/sports.py

```diff
@@ -0,0 +1,133 @@
+#   - in the off-season, this will return info on the latest game played.
+#   input_parameter specifies which type of information to get depending on what is entered.
+#   - full_name is something like "Pittsburgh Panthers vs.  Penn State Nittany Lions"
+#   - short_name is something like "PITT vs. PSU"
+#   - season_name is something like "Regular Season", "Playoffs", "Pre-season", etc.
+#   - week is something like "Week 19"
+#   - court returns the location of the game being played.
+
+# get_basketball_standings simply returns a string resembling something like "14th in ACC"
+
+
+def get_basketball_record() -> str:
+    try:
+        record_summary = basketball_data['team']['record']['items'][0]['summary']
+
+    except KeyError:
+        record_summary = "There's no record right now."
+
+    return record_summary
+
+
+def get_next_basketball_game(input_parameter: str) -> str:
+
+    return_value = "Error: Input parameter"
+
+    if input_parameter == "full_name":
+        return_value = basketball_data['team']['nextEvent'][0]['name']
+    if input_parameter == "short_name":
+        return_value = basketball_data['team']['nextEvent'][0]['shortName']
+
+    if input_parameter == "season_name":
+        return_value = basketball_data['team']['nextEvent'][0]['seasonType']['name']
+
+    if input_parameter == "week":
+        return_value = basketball_data['team']['nextEvent'][0]['week']['text']
+
+    if input_parameter == "court":
+        return_value = basketball_data['team']['nextEvent'][0]['competitions'][0]['venue']['fullName']
+
+    return return_value
+
+
+def get_basketball_standings() -> str:
+    return_value = basketball_data['team']['standingSummary']
+    return return_value
+
+
+# end basketball methods
+
+
+# football methods
+# get_football_record() is a method designed to get the current record of the Men's basketball team.
+#   - During the off-season this data should return "There's no record right now" TODO: ADD THIS EXCEPTION CATCH
+
+# get_next_football_game(input_parameter: str) is a method to get specific info on the next matchup.
+#   - in the off-season, this will return info on the latest game played.
+# input_parameter specifies which type of information to get depending on what is entered.
+#   - full_name is something like "Pittsburgh Panthers vs.  Penn State Nittany Lions"
+#   - short_name is something like "PITT vs. PSU"
+#   - season_name is something like "Regular Season", "Playoffs", "Pre-season", etc.
+#   - week is something like "Week 19"
+#   - field returns the location of the game being played.
+
+# get_basketball_standings simply returns a string resembling something like "14th in ACC"
+
+
+def get_football_record() -> str:
+    record_summary = football_data['team']['record']['items'][0]['summary']
+    return record_summary
+
+
+def get_next_football_game(input_parameter: str) -> str:
+
+    return_value = "Error: Input parameter"
+
+    if input_parameter == "full_name":
```

Comment:
So I have an issue with this `input_parameter` concept. It forces the user to memorize concepts for just this one function which I find not necessary.  It would probably be more beneficial just to return a dict kind of like this one below because the user will get the structure and will have easy knowledge of the keys for each value.

```python
return_value = {
    'full_name': return_value = football_data['team']['nextEvent'][0]['name'],
    'short_name': football_data['team']['nextEvent'][0]['shortName'],
     ...
}
```

Reply:
+1, especially considering future extensibility: Different sports are going to have different data points of interest which will be expressed differently. Take advantage of the opportunity to normalize that into a consistent object (like a dict) that someone would be able to use to programmatically make some direct comparisons, e.g,. "which of the sports teams plays next?"

(That said, without looking too closely at the ESPN data, I suspect it's fairly normalized already for their corporate purposes.)

Bonus is that it only requires one call to the external API once if multiple data points about a team are needed.